scripts/buildWebtypes.js

```diff
@@ -0,0 +1,235 @@
+const path = require('path');
+const fs = require('fs');
+const os = require('os');
+const { execSync } = require('child_process');
+
+const PLAIN_WEB_TYPES_FILE = 'web-types.json';
+const LIT_WEB_TYPES_FILE = 'web-types.lit.json';
+
+const blacklistedPackages = [
+  /^vaadin-/,
+  /^component-base/,
+  /^field-base/,
+  /^field-highlighter/,
+  /^icons/,
+  /^input-container/,
+  /^lit-renderer/,
+  /^polymer-legacy-adapter/,
+];
+
+// Additional attributes that will be added to all components
+const additionalAttributes = [
+  // Theme attribute is not properly declared at the moment
+  {
+    name: 'theme',
+    description: 'The theme variants to apply to the component.',
+    type: 'string | null | undefined',
+  },
+];
+
+/**
+ * Get packages using lerna, excluding blacklisted packages
+ */
+function getRelevantPackages() {
+  const output = execSync('./node_modules/.bin/lerna ls --json --loglevel silent');
+  const allPackages = JSON.parse(output.toString()).map((project) => project.name.replace('@vaadin/', ''));
+  return allPackages.filter((pkg) => !blacklistedPackages.some((blacklistedPackage) => pkg.match(blacklistedPackage)));
+}
+
+function loadAnalysis() {
+  const analysisPath = path.resolve('./analysis.json');
+  try {
+    return JSON.parse(fs.readFileSync(analysisPath, 'utf8'));
+  } catch (e) {
+    throw new Error(
+      `Could not read output of the Polymer Analyzer from: ${analysisPath}. Make sure to run the Polymer Analyzer before generating web-types.`,
+    );
+  }
+}
+
+/**
+ * Maps a type string such as `'boolean | null | undefined'` into an array of type
+ * strings, such as `['boolean', 'null', 'undefined']`
+ * @param typeString
+ * @returns {string[]}
+ */
+function mapType(typeString) {
+  const sanitizedTypeString = (typeString || '').replace(/[!()]/g, '');
+  const types = sanitizedTypeString.split('|');
+  return types.map((type) => type.trim());
+}
+
+function createPlainElementDefinition(elementAnalysis) {
+  const attributes = [...elementAnalysis.attributes, ...additionalAttributes].map((attribute) => ({
```

Comment:
Looks like attributes also need to be filtered to exclude those marked as `readOnly`. Example:
![readonly-attribute](https://user-images.githubusercontent.com/10589913/176860906-8ba08006-5212-4a6d-adc0-ec978820706f.png)

https://github.com/vaadin/web-components/blob/06e5875be93ca50da2846dafc65a8531010c0576/packages/context-menu/src/vaadin-context-menu.js#L251-L256

Reply:
Added a filter that looks up the corresponding property and checks that the property is not read-only.